vembrane/modules/tag.py

```diff
@@ -1,8 +1,9 @@
 import re
 import sys
+from collections.abc import Iterator
 from itertools import chain, islice
 from sys import stderr
-from typing import Dict, Iterator, Set, Tuple
+from types import MappingProxyType
```

Comment:
Just curious: What are these `MappingProxyType`s?

Reply:
Basically a read-only dict. In our case, these are mainly used as wrappers of empty dicts used as default keyword args (which supply additional, fixed information, such as `auxiliary`). For example, when you invoke `filter(..., auxiliary=foo)` with `foo = {}`, you could mutate `foo` inside of `filter`, potentially causing unwanted behaviour/side-effects etc. Therefore, we wrap that empty dict in a `MappingProxyType`, so no new keys etc can be inserted.